packages/alfa-dom/src/common/predicate.ts

```diff
@@ -0,0 +1,32 @@
+import { Predicate } from "@siteimprove/alfa-predicate";
+import {Element, Namespace, Node} from "..";
+
+const { and, equals, property } = Predicate;
+
+export function hasNamespace(
+  predicate: Predicate<Namespace> = () => true
+): Predicate<Element> {
+  return (element) => element.namespace.some(predicate);
+}
+
+export function hasName<T extends { readonly name: string }>(
+  predicate: Predicate<string> = () => true
+): Predicate<T> {
+  return property("name", predicate);
+}
+
+export function isElementByName(
+  ...names: Array<string>
+): Predicate<Node, Element> {
+  return and(
+    Element.isElement,
+    and(hasNamespace(equals(Namespace.HTML)), hasName(equals(...names)))
+  );
+}
+
+export function isDescendantOf(
+  node: Node,
+  options?: Node.Traversal
+): Predicate<Node> {
+  return (desc) => node.descendants(options).find(equals(desc)).isSome();
+}
```

Comment:
This would be well-suited as `Node#includes(descendant: Node, options?: Node.Traversal): boolean` and `Node.includes(descendant: Node, options?: Node.Traversal): Predicate<Node>`.

Reply:
Seeing how this actually looks for the inverse relation, I'd suggest both `hasDescendant()` and `hasAncestor()` as well.